include/rcutils/macros.h

```diff
@@ -131,6 +131,56 @@ extern "C"
 # define RCUTILS_UNLIKELY(x) (x)
 #endif  // _WIN32
 
+#if defined RCUTILS_ENABLE_FAULT_INJECTION
+#include "rcutils/testing/fault_injection.h"
+
+/**
+ * \def RCUTILS_CAN_RETURN_WITH_ERROR_OF
+ * Indicating macro that the function intends to return possible error value.
+ *
+ * Put this macro as the first line in the function. For example:
+ *
+ * int rcutils_function_that_can_fail() {
+ *   RCUTILS_CAN_RETURN_WITH_ERROR_OF(RCUTILS_RET_INVALID_ARGUMENT);
+ *   ...  // rest of function
+ * }
+ *
+ * For now, this macro just simply calls `RCUTILS_FAULT_INJECTION_MAYBE_RETURN_ERROR` if fault
+ * injection is enabled. However, for source code, the macro annotation
+ * `RCUTILS_CAN_RETURN_WITH_ERROR_OF` helps clarify that a function may return a value signifying
+ * an error and what those are.
```

Comment:
@brawner meta, thinking out loud: a string literal that describes the situation in which said error can be returned would be valuable for documentation and could even be printed and/or logged e.g.:

```
RCUTILS_CAN_RETURN_WITH_ERROR_OF(nullptr, "on failed publisher name allocation");
RCUTILS_CAN_RETURN_WITH_ERROR_OF(nullptr, "on failed rmw publisher initialization");
```

It also sets the scene for future handling of side-effects.